<launch>
    <!-- Arguments -->
    <!-- Run rviz and simulator. Launch with "visualize:=true" to enable-->
    <arg name="visualize" default="false"/>
    <!-- <arg name="cam_frame" default="camera_frame" /> -->

    <node name="odom_node" pkg="quack_norris" type="odom.py" output="screen"/>
    <node pkg="robot_localization" type="ekf_localization_node" name="ekf_localization" output="screen">
        <rosparam command="load" file="$(find quack_norris)/config/ekf_config.yaml"/>
    </node>
    <node pkg="quack_norris" type="main_node.py" name="main_node" output="screen" />

    <rosparam command="load" file="$(find quack_norris)/params/apriltags.yaml" ns="apriltags" />

    <!-- Launch the local planner node -->
    <!-- <node pkg="quack_norris" type="local_planner.py" name="local_planner_node" output="screen" /> -->
    <rosparam command="load" file="$(find quack_norris)/params/tag_map.yaml" ns="tag_map" />

    
    <!-- Launch the apriltags2_ros Node -->
    <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="apriltags" 
          clear_params="true" output="screen" launch-prefix="" required="true">
        <remap from="image_rect" to="/$(env VEHICLE_NAME)/camera_node/rect" />
        <remap from="camera_info" to="/$(env VEHICLE_NAME)/camera_node/camera_info" />
        <!-- <param name="camera_frame" type="str" value="$(arg cam_frame)" /> -->
        
        <param name="publish_tag_detections_image" type="bool" value="true" /> 
        <param name="publish_tf" type="bool" value="true" />
        <param name="tag_family" type="str" value="tag36h11" />
        <param name="tag_border" type="int" value="1" />
        <param name="tag_threads" type="int" value="2" />
        <param name="tag_decimate" type="double" value="1.0" />
        <param name="tag_blur" type="double" value="0.0" />
        <param name="tag_refine_edges" type="bool" value="true" />
        <param name="tag_refine_decode" type="bool" value="false" />
        <param name="tag_refine_pose" type="bool" value="false" />
        <param name="tag_debug" type="bool" value="true" />
    </node>

    <group if="$(arg visualize)">
        <include file="$(find quack_norris)/launch/rviz.launch"/>
    </group>
</launch>